mitm.go

```diff
@@ -115,7 +116,27 @@ func (p *Proxy) serveConnect(w http.ResponseWriter, r *http.Request) {
 }
 
 func (p *Proxy) cert(names ...string) (*tls.Certificate, error) {
-	return genCert(p.CA, names)
+	var (
+		cache = map[string]tls.Certificate{} // names (joined with ,) -> cert
```

Comment:
It looks like the cache is in a local variable and thrown away when this method returns.